cpp/src/communicator/mpi.cpp

```diff
@@ -151,7 +156,10 @@ std::unique_ptr<Communicator::Future> MPI::recv(
 std::pair<std::unique_ptr<std::vector<uint8_t>>, Rank> MPI::recv_any(Tag tag) {
     int msg_available;
     MPI_Status probe_status;
-    RAPIDSMPF_MPI(MPI_Iprobe(MPI_ANY_SOURCE, tag, comm_, &msg_available, &probe_status));
+    MPI_Message matched_msg;
+    RAPIDSMPF_MPI(MPI_Improbe(
+        MPI_ANY_SOURCE, tag, comm_, &msg_available, &matched_msg, &probe_status
+    ));
```

Comment:
To do the same change in the UCXX communicator requires https://github.com/rapidsai/ucxx/issues/457